src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -462,6 +500,90 @@ public function getEcotax()
         return null;
     }
 
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getManufacturerName()
+    {
+        if (!isset($this->product['manufacturer_name'])) {
+            // Assign empty value
+            $this->product['manufacturer_name'] = null;
+
+            // If we have manufacturer ID, we will try to load it's name and assign it
+            if (!empty($this->product['id_manufacturer'])) {
+                $manufacturerName = Manufacturer::getNameById((int) $this->product['id_manufacturer']);
+                if (!empty($manufacturerName)) {
+                    $this->product['manufacturer_name'] = $manufacturerName;
+                }
+            }
+        }
+
+        return $this->product['manufacturer_name'];
+    }
+
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getCategory()
+    {
+        if (!isset($this->product['category'])) {
+            $categoryLinkRewrite = Category::getLinkRewrite(
+                (int) $this->product['id_category_default'], (int) $this->language->id
+            );
+            $this->product['category'] = !empty($categoryLinkRewrite) ? $categoryLinkRewrite : null;
+        }
+
+        return $this->product['category'];
+    }
+
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getCategoryName()
+    {
+        if (!isset($this->product['category_name'])) {
+            $categoryName = (string) Db::getInstance()->getValue(
+                'SELECT name FROM ' . _DB_PREFIX_ . 'category_lang 
+                WHERE id_shop = ' . (int) Context::getContext()->shop->id
+                . ' AND id_lang = ' . (int) $this->language->id
+                . ' AND id_category = ' . (int) $this->product['id_category_default']
+            );
+            $this->product['category_name'] = !empty($categoryName) ? $categoryName : null;
+        }
```

Comment:
```suggestion
        if (!isset($this->product['category_name'])) {
            $categoryObject = new Category((int) $this->product['id_category_default'], null, (int) Context::getContext()->shop->id);
            $this->product['category_name'] = $categoryObject->getName((int) $this->language->id);
        }
```

### NOTE :
 I suggest to add a static cache for category object

Reply:
Well, we could do this, but there is a slight additional performance overhead, checking if category image exists, whole object construct etc.

I checked it in the current form and the Category objects are not constructed during product listings. So if somebody uses this property, it would be the first time.

I thought that maybe constructing the Category object would be good if we could reuse it... but the only place we need categories in product listings is for URL rewrite. And the rewrite is using `Category::getLinkRewrite` method, which has it's own caching and is faster than constructing Category objects.